pkg/storage/ingest/reader.go

```diff
@@ -110,13 +113,14 @@ func newPartitionReader(kafkaCfg KafkaConfig, partitionID int32, instanceID stri
 		partitionID:                           partitionID,
 		newConsumer:                           consumer,
 		consumerGroup:                         kafkaCfg.GetConsumerGroup(instanceID, partitionID),
-		metrics:                               newReaderMetrics(partitionID, reg),
 		consumedOffsetWatcher:                 newPartitionOffsetWatcher(),
 		concurrentFetchersMinBytesMaxWaitTime: defaultMinBytesMaxWaitTime,
 		logger:                                log.With(logger, "partition", partitionID),
 		reg:                                   reg,
 	}
 
+	r.metrics = newReaderMetrics(partitionID, reg, func() float64 { return r.BufferedRecords() })
```

Comment:
[nit] Following my suggestion to change `BufferedRecords()` to return an `int64`, then we would have to change this to:

```suggestion
	r.metrics = newReaderMetrics(partitionID, reg, func() float64 { return float64(r.BufferedRecords()) })
```

Reply:
Done in https://github.com/grafana/mimir/pull/9855/commits/43f6386c4e650ca5b22d651d4ba0a7cb60d9672d